### PR TITLE
:tada:  Source Mixpanel: Beta preparation

### LINKED ISSUE #5355: 🐛 Source mixpanel: basic normalization fails with duplicate field names
## Enviroment
- **Airbyte version**:  0.29.4-alpha
- **OS Version / Instance**: macOS
- **Deployment**:  Docker
- **Source Connector and version**: Mixpanel (0.1.0) 
- **Destination Connector and version**: BigQuery (0.3.12)
- **Severity**: High
- **Step where error happened**:  Sync job

## Current Behavior
I’m trying to sync for the first time and the process fails in the normalization step, apparently, something broke the dbt process, the log is attached.
If I try without normalization (raw data), it runs successfully

## Expected Behavior
The normalized data appears at BigQuery without any error

## Logs

[logs-10-0.txt](https://github.com/airbytehq/airbyte/files/6970980/logs-10-0.txt)

**UPDATE - Apparently I have a field called username and another one userName, when I check the destination_catalog.json, I can see both fields, maybe this is causing the ambiguity that BigQuery is not able to resolve?

## Steps to Reproduce
1. Create a Mixpanel Source Connection and a BigQuery destination
2. Try to sync  with the Basic Normalization selected

### LINKED ISSUE #8427: Source Mixpanel: Sync failure with high volume
## Enviroment
- **Airbyte version**:  0.32.6-alpha
- **OS Version / Instance**: GCP e2-standard-4
- **Deployment**: Docker
- **Source Connector and version**: Mixpanel 0.1.5
- **Destination Connector and version**: BigQuery 0.5.0
- **Severity**: High
- **Step where error happened**: Sync job 

## Current Behavior
When syncing large number of rows from mixpanel to bigquery, the sync just fails in between.
Reducing date window size can help alleviate this problem. However I've noticed that even for data window size of 2, 
if the number of rows in that window is greater than 3 million the same issue occurs.
I suspect its due to OOM as there is a brief spike in memory usage of the python process before its killed.
However the machine has 16 gb of ram, and on an average only around 6 gb was being used before the spike.

Another point of concern is that the ability to configure date window size was removed in MixPanel 0.1.6 and it defaults to 30 days.
Currently the only solution was to reduce date window size which is not possible any more.
Thus OOM is bound to happen as data will be synced for 30 days at once

## Expected Behavior
Sync should succeed regardless of number of rows being synced in the date window

## Logs

<details>
<summary>LOG</summary> 

```

2021-11-25 17:07:37 INFO () DefaultReplicationWorker(lambda$getReplicationRunnable$2):220 - Records read: 1241000
2021-11-25 17:07:38 INFO () DefaultReplicationWorker(lambda$getReplicationRunnable$2):220 - Records read: 1242000
2021-11-25 17:07:38 INFO () DefaultReplicationWorker(lambda$getReplicationRunnable$2):220 - Records read: 1243000
2021-11-25 17:08:52 INFO () DefaultReplicationWorker(run):138 - Source thread complete.
destination - 2021-11-25 17:08:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:08:52 INFO i.a.i.b.FailureTrackingAirbyteMessageConsumer(close):60 - {} - Airbyte message consumer: succeeded.
2021-11-25 17:08:52 INFO () DefaultReplicationWorker(run):139 - Waiting for destination thread to join.
destination - 2021-11-25 17:08:52 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:08:52 INFO i.a.i.d.b.BigQueryRecordConsumer(close):163 - {} - Started closing all connections
destination - 2021-11-25 17:08:53 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:08:53 INFO i.a.i.d.b.BigQueryRecordConsumer(closeNormalBigqueryStreams):278 - {} - Waiting for jobs to be finished/closed
destination - 2021-11-25 17:09:02 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:09:02 INFO i.a.i.d.b.BigQueryRecordConsumer(closeNormalBigqueryStreams):295 - {} - Replication finished with no explicit errors. Copying data from tmp tables to permanent
destination - 2021-11-25 17:09:06 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:09:06 INFO i.a.i.d.b.BigQueryRecordConsumer(copyTable):363 - {} - successfully copied table: GenericData{classInfo=[datasetId, projectId, tableId], {datasetId=vayu_staging, tableId=_airbyte_tmp_xvo_mixpanel_export}} to table: GenericData{classInfo=[datasetId, projectId, tableId], {datasetId=vayu_staging, tableId=_airbyte_raw_mixpanel_export}}
2021-11-25 17:09:06 INFO () DefaultReplicationWorker(lambda$getDestinationOutputRunnable$3):248 - state in DefaultReplicationWorker from Destination: io.airbyte.protocol.models.AirbyteMessage@72408d4f[type=STATE,log=<null>,spec=<null>,connectionStatus=<null>,catalog=<null>,record=<null>,state=io.airbyte.protocol.models.AirbyteStateMessage@4fc98083[data={"export":{"date":"2019-03-03T23:59:59"}},additionalProperties={}],additionalProperties={}]
destination - 2021-11-25 17:09:06 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:09:06 INFO i.a.i.d.b.BigQueryRecordConsumer(closeNormalBigqueryStreams):312 - {} - Removing tmp tables...
destination - 2021-11-25 17:09:06 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:09:06 INFO i.a.i.d.b.BigQueryRecordConsumer(closeNormalBigqueryStreams):314 - {} - Finishing destination process...completed
destination - 2021-11-25 17:09:06 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:09:06 INFO i.a.i.b.IntegrationRunner(run):133 - {} - Completed integration: io.airbyte.integrations.destination.bigquery.BigQueryDestination
destination - 2021-11-25 17:09:06 INFO () DefaultAirbyteStreamFactory(lambda$create$0):61 - 2021-11-25 17:09:06 INFO i.a.i.d.b.BigQueryDestination(main):336 - {} - completed destination: class io.airbyte.integrations.destination.bigquery.BigQueryDestination
2021-11-25 17:09:06 INFO () DefaultReplicationWorker(run):141 - Destination thread complete.
2021-11-25 17:09:06 ERROR () DefaultReplicationWorker(run):145 - Sync worker failed.

```

</details>

## Steps to Reproduce
1. Setup Mixpanel source and Bigquery destination
2. Set up the connection to output only raw data (without normalization)
3. Need to ensure a large number of rows are being synced in each date window. (This can be achieved by either setting a large date window, or allocating less memory)

## Are you willing to submit a PR?
Yes, but might take time as I am new to this.

### LINKED ISSUE #9410: :bug: Source Mixpanel: 429 status code is not handled
*Is this your first time deploying Airbyte*: No
*OS Version / Instance*:  EC2 t2.medium
*Deployment*: Docker
*Airbyte Version*: 0.35.2-alpha
*Source name/version*: Mixpanel 0.1.9
*Destination name/version*: AWS Redshift
*Description*: Hi everyone, I am setting up a connection from Mixpanel to Redshift. The data appears to be loaded into Redshift properly, but one error is logged all the time when I sync. Does anybody know where it comes from?

When reading stream `SourceMixpanel` the connector does not handle `429` response code properly, leading to `DefaultBackoffException`. The connector should override `backoff_time` for smarter rate limit handling.

## Error
```
Stream funnels: 429 Too Many Requests - {"request": "/api/2.0/funnels?from_date=2021-03-22&to_date=2021-04-20&funnel_id=15495008&unit=day", "error": "Query rate limit exceeded for project_id: 2526451. 1/5 queries running concurrently and 61/60 queries running in the last hour. For more information, please consult our documentation at https://help.mixpanel.com/hc/en-us/articles/115004602563-Rate-Limits-for-Export-API-Endpoints"}

Encountered an exception while reading stream SourceMixpanel
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 108, in read
    internal_config=internal_config,
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 141, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 185, in _read_incremental
    for record_counter, record_data in enumerate(records, start=1):
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/streams/http/http.py", line 352, in read_records
    response = self._send_request(request, request_kwargs)
  File "/airbyte/integration_code/source_mixpanel/source.py", line 78, in _send_request
    raise e
  File "/airbyte/integration_code/source_mixpanel/source.py", line 73, in _send_request
    return super()._send_request(request, request_kwargs)
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/streams/http/http.py", line 319, in _send_request
    return backoff_handler(user_backoff_handler)(request, request_kwargs)
  File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/streams/http/http.py", line 282, in _send
    raise DefaultBackoffException(request=request, response=response)
```

## Logs
[logs-34-0 (1).txt](https://github.com/airbytehq/airbyte/files/7847293/logs-34-0.1.txt)

[Slack conversation](https://airbytehq.slack.com/archives/C01MFR03D5W/p1641851090171100?thread_ts=1641851090.171100&cid=C01MFR03D5W)

### PR DESCRIPTION
**Pull request by roman-romanov-o**
## What
1. Acceptances tests were failing owing to [changed ratelimit](https://help.mixpanel.com/hc/en-us/articles/115004602563-Rate-Limits-for-API-Endpoints) \(60 requests per hour for any endpoint\)
2. [Ratelimit issue](https://github.com/airbytehq/airbyte/issues/9410) & https://github.com/airbytehq/alpha-beta-issues/issues/70
3. [High volume issue](https://github.com/airbytehq/airbyte/issues/8427)
4. [Normaliztion failed for custom Export](https://github.com/airbytehq/airbyte/issues/5355)
5. Move streams to distinct module to reduce file size
6. Update doc

## How
1. Added decorator for `streams`, that check `is_testing` flag in config, and reduces amount of requests that is made in acceptance tests
2. Reduce ratelimit for all connectors
3. Add `date_window_size` to spec, so client will be able to iterate through data via smaller batches
4. 
Sort property names, and check for such cases, and transform `userName` -> `_userName`

If such collission occurs more than one time, then
```
userName
UserName
username
```
will be transformed to 
```
userName
_UserName
__username
```

## Recommended reading order

Huge line change number is caused only by moving streams into distinct module

* Only real change in `source.py` is added decorator to `streams` method
* Only real change in sources is updated `Export` (streams/export.py) source, where properties are transformed via `transform_property_names`

## Pre-merge Checklist
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]
LGTM

**Review by girarda — APPROVED** [review]

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    API Rate Limit Handler: after each request freeze for the time period: 3600/reqs_per_hour_limit seconds
+    """
+
+    @property
+    def url_base(self):
+        prefix = "eu." if self.region == "EU" else ""
+        return f"https://{prefix}mixpanel.com/api/2.0/"
+
+    # https://help.mixpanel.com/hc/en-us/articles/115004602563-Rate-Limits-for-Export-API-Endpoints#api-export-endpoint-rate-limits
+    reqs_per_hour_limit: int = 60  # 1 query per minute
```
[the remark this answers is not in the document]
Comment (roman-romanov-o) [reply airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:30]: Old limit was 400, but now they changed it to 60

I fixed the comment

**Review by bazarnov — APPROVED** [review]

**Comment by supertopher** [comment]
@bazarnov This PR is breaking production builds because it has not been published

stack trace

```
2022-06-03 12:54:33 INFO i.a.c.s.SeedConnectorSpecGenerator(fetchSpecFromGCS):119 - Seeded spec not found for docker image airbyte/source-mixpanel:0.1.17 - fetching from GCS bucket io-airbyte-cloud-spec-cache...
Exception in thread "main" java.lang.RuntimeException: Failed to fetch valid spec file for docker image airbyte/source-mixpanel:0.1.17 from GCS bucket io-airbyte-cloud-spec-cache. This will continue to fail until the connector change has been approved and published. See https://github.com/airbytehq/airbyte/tree/master/docs/connector-development#publishing-a-connector for more details.
        at io.airbyte.config.specs.SeedConnectorSpecGenerator.lambda$fetchSpecFromGCS$4(SeedConnectorSpecGenerator.java:121)
        at java.base/java.util.Optional.orElseThrow(Optional.java:403)
        at io.airbyte.config.specs.SeedConnectorSpecGenerator.fetchSpecFromGCS(SeedConnectorSpecGenerator.java:121)
        at io.airbyte.config.specs.SeedConnectorSpecGenerator.lambda$fetchUpdatedSeedSpecs$3(SeedConnectorSpecGenerator.java:114)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at io.airbyte.config.specs.SeedConnectorSpecGenerator.fetchUpdatedSeedSpecs(SeedConnectorSpecGenerator.java:115)
        at io.airbyte.config.specs.SeedConnectorSpecGenerator.run(SeedConnectorSpecGenerator.java:83)
        at io.airbyte.config.specs.SeedConnectorSpecGenerator.main(SeedConnectorSpecGenerator.java:73)

> Task :airbyte-config:specs:generateSeedConnectorSpecs FAILED
```

I honestly don't really understand the workflow or what publishing is exactly in this use case.  Can you help me understand the best way to get prod back in fighting shape quickly?

**Pull request merged**